appengine_backend.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2018 Adam Tauber
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package colly
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/gob"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	"google.golang.org/appengine/urlfetch"
+)
+
+type appEngineBackend struct {
+	LimitRules []*LimitRule
+	Client     *http.Client
+	lock       *sync.RWMutex
+}
+
+func (h *appEngineBackend) Init(jar http.CookieJar) {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Is it necessary to set the seed here?

Reply:
:thinking:  probably not, it was a copy and paste from the current `http_backend`

Reply:
Good, I dunno if this has been fixed, but until a couple years ago [rand wasnt safe for concurrent use](https://blog.sgmansfield.com/2016/01/the-hidden-dangers-of-default-rand/) , not sure if it affects the seed